src/backend/distributed/master/master_stage_protocol.c

```diff
@@ -570,6 +569,24 @@ CreateShardsOnWorkersViaExecutor(Oid distributedRelationId, List *shardPlacement
 		taskList = lappend(taskList, task);
 	}
 
+	if (useExclusiveConnection)
+	{
+		/*
+		 * When the table has local data, we force max parallelization so data
+		 * copy is done efficiently.
+		 */
+		SetLocalForceMaxQueryParallelization();
+
+		/*
+		 * TODO: we shouldn't need to set poolSize when setting ForceMaxQueryParallelization.
+		 *
```

Comment:
so should we remove this or ...?

Reply:
I updated the comment to make it more comprehensible. Currently adaptive executor ignores ForceMaxQueryParallelization for recording parallel access and uses poolSize > 1 check instead. Making adaptive executor to take into account ForceMaxQueryParallelization isn't as easy as thought, since it seems that then some TRUNCATE related issues arise. So we just set poolSize here to force recording parallel access, which is a hacky way to do it, and should be removed in future once we have the proper checks in adaptive executor.